main/src/ca/uwaterloo/flix/language/errors/RedundancyError.scala

```diff
@@ -338,6 +340,38 @@ object RedundancyError {
     })
   }
 
+  /**
+    * An error raised to indicate that a function expression is useless.
+    *
+    * @param tpe the type of the expression.
+    * @param loc the location of the expression.
+    */
+  case class UselessFunctionExpression(tpe: Type, loc: SourceLocation) extends RedundancyError {
```

Comment:
UselessFunctionExpression -> UnderAppliedFunction

Reply:
So the useless part is left implicit? I think the name implies that it is an error to partially apply something